cmd/migrate/workers_script.go

```diff
@@ -0,0 +1,233 @@
+	block.Body().RenameAttribute("name", "script_name")
+
+	// Transform v4 binding blocks to v5 bindings list
+	transformBindings(block, diags)
+}
+
+// transformWorkersScriptStateJSON transforms the state JSON for workers_script
+// V4: name -> V5: script_name
+// Also transforms v4 binding attributes to v5 bindings list
+func transformWorkersScriptStateJSON(jsonStr string, path string) string {
+	result := jsonStr
+
+	// Get the current name value
+	namePath := path + ".attributes.name"
+	scriptNamePath := path + ".attributes.script_name"
+
+	// Get the value and move it
+	nameValue := gjson.Get(jsonStr, namePath)
+	if nameValue.Exists() {
+		// Set the new attribute
+		result, _ = sjson.Set(result, scriptNamePath, nameValue.Value())
+		// Delete the old attribute
+		result, _ = sjson.Delete(result, namePath)
+	}
+
+	// Transform v4 binding attributes to v5 bindings list
+	result = transformBindingsInState(result, path)
+
+	// Remove unsupported attributes that exist in v4 but not in v5
+	dispatchNamespacePath := path + ".attributes.dispatch_namespace"
+	result, _ = sjson.Delete(result, dispatchNamespacePath)
+
+	// Remove hyperdrive_config_binding which exists in v4 but not expected in v5 schema
+	hyperdriveConfigBindingPath := path + ".attributes.hyperdrive_config_binding"
+	result, _ = sjson.Delete(result, hyperdriveConfigBindingPath)
+
+	// Remove module attribute which exists in v4 but causes issues in v5
+	modulePath := path + ".attributes.module"
+	result, _ = sjson.Delete(result, modulePath)
+
+	// Fix placement attribute format: v4 uses array [], v5 expects object {}
+	placementPath := path + ".attributes.placement"
+	placementValue := gjson.Get(result, placementPath)
+	if placementValue.Exists() && placementValue.IsArray() {
+		// If placement is an empty array, set it to empty object
+		if len(placementValue.Array()) == 0 {
+			result, _ = sjson.Set(result, placementPath, map[string]interface{}{})
+		}
+	}
+
+	return result
+}
+
+// transformBindingsInState converts v4 binding attributes to v5 bindings format in state JSON
+func transformBindingsInState(jsonStr string, path string) string {
+	result := jsonStr
+
+	// List of v4 binding attribute names to look for
+	v4BindingAttributes := []string{
+		"plain_text_binding",
+		"kv_namespace_binding",
+		"secret_text_binding",
+		"r2_bucket_binding",
+		"queue_binding",
+		"d1_database_binding",
+		"analytics_engine_binding",
+		"service_binding",
+		"webassembly_binding",
+	}
+
+	var bindings []interface{}
+
+	// Check each v4 binding type and convert to v5 bindings format
+	for _, bindingAttr := range v4BindingAttributes {
+		bindingPath := path + ".attributes." + bindingAttr
+		bindingValue := gjson.Get(jsonStr, bindingPath)
+
+		if bindingValue.Exists() {
+			// Convert binding attribute name to binding type
+			bindingType := bindingAttr
+			if bindingType[len(bindingType)-8:] == "_binding" {
+				bindingType = bindingType[:len(bindingType)-8]
+			}
```

Comment:
slicing off the `_binding` suffix isn't sufficient in all cases, e.g. `d1_database_binding` attributes are now bindings of type `"d1"` in v5, and `hyperdrive_config_binding` => `"hyperdrive"`. those should be the only exceptions though.